docs: document SQLDatabase.named for Encore.ts (#1588)
encoredev/encore: docs/ts/primitives/databases.md

For more information on using ORMs with Encore, see the [ORMs](/docs/ts/develop/orms) page.

## Sharing databases between services

> Encore tracks which databases are used by which services. This allows Encore's Cloud Platform
> to automatically manage permissions and set up database credentials for the services that use
> each database.
> 
> In order to share a database between services, you have two options.
> 
> You can define the `SQLDatabase` object in a shared module as an exported variable, and reference this object
> from every service that needs to access the database.
> 
> Or, alternatively, you can have define the `SQLDatabase` object inside one of the services using `new SQLDatabase("name", ...)`,
> and have the other services that need to access it create their own reference using `SQLDatabase.named("name")`.
> 
> Both of these approaches have the same effect, but the latter is more explicit.

Reviewer: ```suggestion
There are two primary ways of sharing a database between services:

- You can define the `SQLDatabase` object in a shared module as an exported variable, and reference this object
from every service that needs to access the database.
- You can define the `SQLDatabase` object in one service using `new SQLDatabase("name", ...)`, and have other services access it by creating a reference using `SQLDatabase.named("name")`.

Both approaches have the same effect, but the latter is more explicit.
```
